full-stack-tests/presentation/src/learning-snippets/Hierarchies.test.ts

```diff
@@ -24,305 +28,1618 @@ describe("Learning Snippets", () => {
+              hideIfNoChildren: true,
+            }],
+          }, {
+            ruleType: RuleTypes.ChildNodes,
+            condition: `ParentNode.IsOfClass("PhysicalModel", "BisCore")`,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.CustomNode,
+              type: "child",
+              label: "Child",
+            }],
+          }],
+        };
+        // __PUBLISH_EXTRACT_END__
+        printRuleset(ruleset);
+
+        // Verify that only PhysicalModel is displayed
+        const classGroupingNodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset });
+        expect(classGroupingNodes).to.have.lengthOf(1).and.to.containSubset([{
+          label: { displayValue: "Physical Model" },
+        }]);
+
+        const modelNodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset, parentKey: classGroupingNodes[0].key });
+        expect(modelNodes).to.have.lengthOf(1).and.to.containSubset([{
+          label: { displayValue: "Properties_60InstancesWithUrl2" },
+        }]);
+
+        const customNodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset, parentKey: modelNodes[0].key });
+        expect(customNodes).to.have.lengthOf(1).and.to.containSubset([{
+          label: { displayValue: "Child" },
+        }]);
+      });
+
+      it("uses `hideExpression` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ Hierarchies.Specification.HideExpression.Ruleset
+        // The ruleset contains a root node specification for `bis.GroupInformationModel` and `bis.PhysicalModel` nodes which
+        // are grouped by class and hidden if there's a children artifact "isTestNode". The artifact is created only
+        // for `bis.GroupInformationModel` node by its custom child node. This makes the `bis.GroupInformationModel` node replaced by
+        // its children.
+        const ruleset: Ruleset = {
+          id: "example",
+          rules: [{
+            ruleType: RuleTypes.RootNodes,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.InstanceNodesOfSpecificClasses,
+              classes: { schemaName: "BisCore", classNames: ["GroupInformationModel", "PhysicalModel"], arePolymorphic: true },
+              hideExpression: `ThisNode.ChildrenArtifacts.AnyMatches(x => x.isTestNode)`,
+            }],
+          }, {
+            ruleType: RuleTypes.ChildNodes,
+            condition: `ParentNode.IsOfClass("GroupInformationModel", "BisCore")`,
+            specifications: [{
+              specType: ChildNodeSpecificationTypes.CustomNode,
+              type: "test",
+              label: "Test Node",
+            }],
+            customizationRules: [{
+              ruleType: RuleTypes.NodeArtifacts,
+              items: {
+                isTestNode: `TRUE`,
+              },
+            }],
           }],
+        };
+        // __PUBLISH_EXTRACT_END__
```

Comment:
Could we concoct a simpler ruleset that does not introduce too many new concepts? Many people won't be familiar with artifacts (certainly I'm not).

Reply:
agree, simplified the example